### PR TITLE
test(mail): add unit tests, supporting cast

### PR DESCRIPTION
**Pull request by LoneRifle**
## Problem

With impending features requested for the next iteration of AskGov, a test framework should be put in place so that
the team is encouraged to provide unit tests along with their implementations. This PR serves to do that, using
MailService as an example.

To facilitate unit-testing of MailService, we also need to do some light refactoring.

## Solution

Change the ctor so that we accept a subset of nodemailer's Transporter. 
This allows us to more easily supply a mock for testing.

- change `mailOptions: SMTPTransport.Options` to
  `transport: Pick<Transporter, 'sendMail'>`
- rework bootstrap to inject an actual nodemailer Transporter

- provide a test for MailService
- include npm run scripts to facilitate testing
  - invoke client tests with `--passWithNoTests` for now since there
    are none
- tighten `server/tsconfig.json` so that it does not pick up files outside the specified `rootDir: src/`
- reinstate test step in GitHub Actions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "postinstall": "npm --prefix server install && npm --prefix client install",
     "auditDep": "npm audit fix && npm --prefix server audit fix && npm --client server audit fix",
     "prepare": "husky install",
+    "test": "npm --prefix server test && CI=true npm --prefix client test -- --passWithNoTests",
```
Comment (nauynix) [inline package.json:17]: Can I check if the -- before --passWithNoTests is required? What does it do?

File: server/src/modules/mail/mail.service.ts [hunk 2]
```diff
@@ -1,18 +1,17 @@
-import { createTransport, Transporter } from 'nodemailer'
-import SMTPTransport from 'nodemailer/lib/smtp-transport'
+import { Transporter } from 'nodemailer'
 import { renderLoginOtpBody } from './mail.util'
 
 export class MailService {
   private mailFromEmail: string
-  private transport: Transporter
+  private transport: Pick<Transporter, 'sendMail'>
   constructor({
```
Comment (nauynix) [inline server/src/modules/mail/mail.service.ts:7]: Theres syntactic sugar for this, credit to @mantariksh.
https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties

File: server/src/modules/mail/mail.service.ts [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
     mailFromEmail,
   }: {
-    mailOptions: SMTPTransport.Options
+    transport: Pick<Transporter, 'sendMail'>
     mailFromEmail: string
   }) {
-    this.transport = createTransport(mailOptions)
+    this.transport = transport
     this.mailFromEmail = mailFromEmail
   }
```
Comment (nauynix) [inline server/src/modules/mail/mail.service.ts:16]: Theres syntactic sugar for this if you want. Credits to @mantariksh 
https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties

File: package.json (hunk 1 above)
Comment (LoneRifle) [reply package.json:17]: `--` passes the arguments after it to the command invoked by the npm run script, so it's needed

**Review by nauynix — APPROVED** [review]
lgtm

**Pull request merged**